tests/e2e_all/scripts/common/common_functions.include.sh

```diff
@@ -552,3 +571,19 @@ buildCurrentCBinaries() {
 }
 
 ### END C ###
+
+setup_type_and_version() {
+  IFS=: read -r type version <<<"$1"
+  case "$type" in
+    d) # dart
+      setupDartVersion "$version" || logErrorAndExit "Failed to set up binaries for dart version [$version]"
+      ;;
+    c) # c
+      setupCVersion "$version" || logErrorAndExit "Failed to set up binaries for c version [$version]"
+      ;;
+    *)
+      logErrorAndExit "This script doesn't know where to find NoPorts daemon binary for [$typeAndVersion]"
+      exit 1
+      ;;
+  esac
+}
```

Comment:
Moved to this file so it could be executed in parallel to build the C and Dart code simultaneously